oioioi/teachers/views.py

```diff
@@ -50,6 +52,18 @@ def is_teacher(request):
 def is_not_teacher(request):
     return not_anonymous(request) and not request.user.has_perm('teachers.teacher')
 
+#nowa rzecz, do autocomplete wyszukiwania, czy to przejdzie?
+def get_usernames(request):
```

Comment:
you can use similar approach as in friend_hints_view in particular you can use oioioi.base.utils.user_selection.get_user_hints_view